agent/util-scripts/pbench-tool-meister-start

```diff
@@ -0,0 +1,450 @@
+#!/usr/bin/env python3
+# -*- mode: python -*-
+
+"""pbench-tool-meister-start
+
+Responsible for:
+
+   1. Starting a redis server
+   2. Loading up the tool group data for the target group into the redis server
+   3. Starting the tool-data-sink process
+   4. Starting all the local and remote tool meisters
+
+When complete we leave running locally, a redis server and a tool data sink
+process, and any local or remote tool meisters.
+
+The pbench-tool-meister-stop will take care of (gracefully) stopping all of
+these process, locally or remotely.
+"""
+
+import sys
+import json
+import os
+import time
+import errno
+import redis
+import logging
+
+# Port number is "One Tool" in hex 0x17001
+# FIXME: move to common area
+redis_port = 17001
+
+# Redis server configuration template for pbench's use
+redis_conf_tmpl = """bind {hostnames}
+daemonize yes
+dir {tm_dir}
+dbfilename pbench-redis.rdb
+logfile {tm_dir}/redis.log
+loglevel notice
+pidfile {tm_dir}/redis_{redis_port:d}.pid
+port {redis_port:d}
+"""
+
+# FIXME: this should be moved to a shared area
+channel = "tool-meister-chan"
+
+
+class ToolGroup(object):
+    tg_prefix = "tools-v1"
+
+    def __init__(self, group):
+        self.group = group
+        _pbench_run = os.environ["pbench_run"]
+        _tg_dir = os.path.join(_pbench_run, "{}-{}".format(self.tg_prefix, self.group))
+        if not os.path.isdir(_tg_dir):
+            raise Exception(
+                "bad tool group, {}: directory {} does not exist".format(group, _tg_dir)
+            )
+        self.tg_dir = os.path.realpath(_tg_dir)
+
+        # __trigger__
+        try:
+            with open(os.path.join(self.tg_dir, "__trigger__"), "r") as fp:
+                _trigger = fp.read()
+        except OSError as ex:
+            if ex.errno != errno.ENOENT:
+                raise
+            # Ignore missing trigger file
+            self.trigger = None
+        else:
+            if len(_trigger) == 0:
+                # Ignore empty trigger file contents
+                self.trigger = None
+            else:
+                self.trigger = _trigger
+
```

Comment:
self.trigger isnt used anywere else?

Reply:
Not at this time.  But it is there so that it will land in the Redis server.